### PR TITLE
Plans 2023: Featured badge for current plan

### PR DESCRIPTION
**Pull request by chriskmnds**
Fixes Automattic/martech#1492

## Proposed Changes

Follow-up to https://github.com/Automattic/wp-calypso/pull/73436 that replaces badges with "your plan" when any one of them is the site's current plan. 

**Notes:** 

- The refactors in [plan-comparison-grid.tsx](https://github.com/Automattic/wp-calypso/pull/73582/files#diff-b28a2a8235dc8fde181278f2cb9c570052550c5276652dc66727943605f37030) may seem daunting, but they mainly moved things out of for-loops and into function components to use hooks. Some duplication around the storage feature row was also cleaned out. Not much other improvement than that (unfortunately).
- Some last-minute changes went into how the top table renders the badges, so should be tested thoroughly with different plans to make sure nothing's missed. 

**Rant:** I'll leave out the rant 💣 

## Media

Screenshot 2023-02-23 at 3 26 03 PM

Screenshot 2023-02-23 at 4 40 55 PM

Screenshot 2023-02-23 at 3 27 19 PM

Screenshot 2023-02-23 at 3 27 32 PM

## Testing Instructions

- go to /plans/[SITE] with ?flags=onboarding/2023-pricing-grid and confirm the changes
- got to /start/plans and confirm the same
- try with different plans and ensure the rounded corners at the top/highlight badges render properly.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/plan-features-2023-grid/style.scss [hunk 1]
```diff
@@ -705,38 +728,6 @@ body.is-section-signup.is-white-signup,
...
-					&.is-premium-plan {
-						border-top-right-radius: 0;
-					}
-					&.is-business-plan {
-						border-top-left-radius: 0;
-					}
-				}
-			}
-		}
-	}
```
Comment (chriskmnds) [inline client/my-sites/plan-features-2023-grid/style.scss:739]: good riddance, I hope

**Review by aneeshd16 — COMMENTED** [review]
Works well, nice refactor! 👍 👍 

The new border introduced for the current plan in this PR overlaps with some of the text in the plans comparison grid:
image
image

Maybe we can reduce the width of those columns to fix this issue?

**Comment by chriskmnds** [comment]
> The new border introduced for the current plan in this PR overlaps with some of the text in the plans comparison grid

Thanks, @aneeshd16. Good catch. The entire first section/column looks a little congested right now.

> Maybe we can reduce the width of those columns to fix this issue?

Sounds reasonable. I think so too. I took a stab at this, but it's turning out more complex than I thought. I will need to persevere longer. I am lost for words, again, working with this grid.

**Comment by chriskmnds** [comment]
Thanks again @aneeshd16 . The issue should be fixed now. I chose to set a `min-width` on the header cells instead, and remove a few other restrictions on the "regular" cells. Thanks for the pointers in Slack. 

A few more changes worth noting:

- It wasn't necessary to remove `max-width: 200px` from the cells to get this working. It just pushed cells a little far to the right while enough space existed in larger viewports. We can revert if it was intentional. 
- Not sure what the deal was with `overflow-x: auto` on the plans wrapper. Leaving it was creating a scroller at the bottom  with all the margins and borders we are adding at places. Removed, but will have another look.

**Review by aneeshd16 — APPROVED** [review]
This works great! Thanks for explaining the changes made. 

There is a code conflict, but it seems to be a minor one.

**Comment by chriskmnds** [comment]
Thanks for taking the time 🏡

**Pull request merged**